core/trino-main/src/main/java/io/trino/sql/ir/IrExpressions.java

```diff
@@ -119,6 +119,9 @@ private static boolean mayFail(ResolvedFunction function)
                 !name.equals(builtinFunctionName("try_cast")) &&
                 !name.equals(builtinFunctionName("$not")) &&
                 !name.equals(builtinFunctionName("substring")) &&
+                !name.equals(builtinFunctionName("trim")) &&
```

Comment:
There's other string functions like `lpad` (and functions other than string functions), what's the criteria for something getting on this list and is there a way to check if we're missing more entries here ?

Reply:
Functions that are guaranteed not to fail for any input should be in this list. 

Eventually, we need to move this to be an attribute of the functions that the optimizer can inspect. So this is a stopgap.

Reply:
Can we add other commonly used string functions like upper, lower, lpad, rpad, reverse etc. too ?
Does it make sense to extend `allow_unsafe_pushdown` to allow bypass this check as well ?

Reply:
Yeah, I'll add other functions.

> Does it make sense to extend allow_unsafe_pushdown to allow bypass this check as well?

No. That check is a temporary workaround for something that has no solution in the current optimizer. For functions, I'd rather add them here.